File: backend/app/analytics/analytics_service.py
"""
Analytics Service für Business Intelligence
"""
import logging
from datetime import datetime, date, timedelta
from typing import Dict, List, Any, Optional, Tuple
from sqlmodel import Session, select, func, and_, or_
from models.analytics import (
    AnalyticsEvent, RevenueMetric, 
    ABTest, EventType, UserAnalytics
)
from models.payment import Payment, PaymentStatus, PaymentMethod
from models.user import User
from models.listing import Listing
import json
import redis
from config import Config

logger = logging.getLogger(__name__)

class AnalyticsService:
    """Service für Analytics und Business Intelligence"""
    
    def __init__(self, session: Session):
        self.session = session
        # Redis-Caching für bessere Performance
        try:
            self.redis_client = redis.from_url(Config.REDIS_URL, decode_responses=True)
            self.redis_client.ping()
            logger.info("✅ Redis-Caching aktiviert")
        except Exception as e:
            logger.warning(f"Redis-Caching deaktiviert: {e}")
            self.redis_client = None
    
    # ==================== CACHING ====================
    
    def _get_cache(self, key: str) -> Optional[Any]:
        """Holt Daten aus Redis-Cache"""
        if not self.redis_client:
            return None
        try:
            cached_data = self.redis_client.get(key)
            if cached_data:
                return json.loads(cached_data)
        except Exception as e:
            logger.warning(f"Cache-Fehler: {e}")
        return None
    
    def _set_cache(self, key: str, data: Any, ttl: int = 300) -> None:
        """Speichert Daten im Redis-Cache"""
        if not self.redis_client:
            return
        try:
            self.redis_client.setex(key, ttl, json.dumps(data, default=str))
        except Exception as e:
            logger.warning(f"Cache-Fehler: {e}")
    
    # ==================== EVENT TRACKING ====================
    
    def track_event(
        self, 
        event_type: EventType, 
        user_id: Optional[int] = None,
        session_id: Optional[str] = None,
        properties: Optional[Dict[str, Any]] = None,
        value: Optional[float] = None,
        ip_address: Optional[str] = None,
        user_agent: Optional[str] = None,
        referrer: Optional[str] = None
    ) -> AnalyticsEvent:
        """Tracke ein Analytics Event"""
        try:
            event = AnalyticsEvent(
                event_type=event_type,
                user_id=user_id,
                session_id=session_id,
                properties=properties or {},
                value=value,
                ip_address=ip_address,
                user_agent=user_agent,
                referrer=referrer
            )
            
            self.session.add(event)
            self.session.commit()
            self.session.refresh(event)
            
            logger.info(f"Event tracked: {event_type} for user {user_id}")
            return event
            
        except Exception as e:
            logger.error(f"Failed to track event {event_type}: {e}")
            self.session.rollback()
            raise
    
    def get_events(
        self,
        event_types: Optional[List[EventType]] = None,
        user_id: Optional[int] = None,
        start_date: Optional[date] = None,
        end_date: Optional[date] = None,
        limit: int = 1000
    ) -> List[AnalyticsEvent]:
        """Hole Events mit Filtern"""
        query = select(AnalyticsEvent)
        
        if event_types:
            query = query.where(AnalyticsEvent.event_type.in_(event_types))
        if user_id:
            query = query.where(AnalyticsEvent.user_id == user_id)
        if start_date:
            query = query.where(AnalyticsEvent.event_date >= start_date)
        if end_date:
            query = query.where(AnalyticsEvent.event_date <= end_date)
        
        query = query.order_by(AnalyticsEvent.created_at.desc()).limit(limit)
        
        return self.session.exec(query).all()
    
    # ==================== REVENUE ANALYTICS ====================
    
    def calculate_daily_revenue(self, target_date: date) -> RevenueMetric:
        """Berechne Tagesumsatz"""
        try:
            # Hole alle Payments für den Tag
            payments = self.session.exec(
                select(Payment).where(
                    and_(
                        func.date(Payment.created_at) == target_date,
                        Payment.status == PaymentStatus.COMPLETED
                    )
                )
            ).all()
            
            # Berechne Metriken
            total_revenue = sum(p.amount for p in payments)
            payment_count = len(payments)
            
            subscription_revenue = sum(
                p.amount for p in payments 
                if p.transaction_type == "subscription"
            )
            one_time_revenue = total_revenue - subscription_revenue
            
            stripe_revenue = sum(
                p.amount for p in payments 
                if p.method == PaymentMethod.STRIPE
            )
            paypal_revenue = sum(
                p.amount for p in payments 
                if p.method == PaymentMethod.PAYPAL
            )
            
            # Hole User-Metriken
            new_users = self.session.exec(
                select(func.count(User.id)).where(
                    func.date(User.created_at) == target_date
                )
            ).first() or 0
            
            active_users = self.session.exec(
                select(func.count(func.distinct(AnalyticsEvent.user_id))).where(
                    and_(
                        AnalyticsEvent.event_date == target_date,
                        AnalyticsEvent.user_id.isnot(None)
                    )
                )
            ).first() or 0
            
            paying_users = len(set(p.user_id for p in payments))
            
            # Erstelle oder aktualisiere RevenueMetric
            existing_metric = self.session.exec(
                select(RevenueMetric).where(RevenueMetric.date == target_date)
            ).first()
            
            if existing_metric:
                existing_metric.total_revenue = total_revenue
                existing_metric.payment_count = payment_count
                existing_metric.subscription_revenue = subscription_revenue
                existing_metric.one_time_revenue = one_time_revenue
                existing_metric.stripe_revenue = stripe_revenue
                existing_metric.paypal_revenue = paypal_revenue
                existing_metric.new_users = new_users
                existing_metric.active_users = active_users
                existing_metric.paying_users = paying_users
                existing_metric.updated_at = datetime.utcnow()
                
                self.session.add(existing_metric)
                self.session.commit()
                return existing_metric
            else:
                metric = RevenueMetric(
                    date=target_date,
                    total_revenue=total_revenue,
                    payment_count=payment_count,
                    subscription_revenue=subscription_revenue,
                    one_time_revenue=one_time_revenue,
                    stripe_revenue=stripe_revenue,
                    paypal_revenue=paypal_revenue,
                    new_users=new_users,
                    active_users=active_users,
                    paying_users=paying_users
                )
                
                self.session.add(metric)
                self.session.commit()
                self.session.refresh(metric)
                return metric
                
        except Exception as e:
            logger.error(f"Failed to calculate daily revenue for {target_date}: {e}")
            self.session.rollback()
            raise
    
    def get_revenue_metrics(
        self,
        start_date: date,
        end_date: date
    ) -> List[RevenueMetric]:
        """Hole Revenue-Metriken für Zeitraum"""
        return self.session.exec(
            select(RevenueMetric).where(
                and_(
                    RevenueMetric.date >= start_date,
                    RevenueMetric.date <= end_date
                )
            ).order_by(RevenueMetric.date)
        ).all()
    
    def get_revenue_summary(self, days: int = 30) -> Dict[str, Any]:
        """Hole Revenue-Zusammenfassung"""
        end_date = date.today()
        start_date = end_date - timedelta(days=days)
        
        metrics = self.get_revenue_metrics(start_date, end_date)
        
        if not metrics:
            return {
                "total_revenue": 0,
                "total_payments": 0,
                "average_daily_revenue": 0,
                "growth_rate": 0,
                "top_payment_method": "N/A"
            }
        
        total_revenue = sum(m.total_revenue for m in metrics)
        total_payments = sum(m.payment_count for m in metrics)
        average_daily_revenue = total_revenue / len(metrics) if metrics else 0
        
        # Wachstumsrate berechnen
        if len(metrics) >= 2:
            first_half = metrics[:len(metrics)//2]
            second_half = metrics[len(metrics)//2:]
            
            first_half_revenue = sum(m.total_revenue for m in first_half)
            second_half_revenue = sum(m.total_revenue for m in second_half)
            
            if first_half_revenue > 0:
                growth_rate = ((second_half_revenue - first_half_revenue) / first_half_revenue) * 100
            else:
                growth_rate = 0
        else:
            growth_rate = 0
        
        # Top Payment Method
        stripe_total = sum(m.stripe_revenue for m in metrics)
        paypal_total = sum(m.paypal_revenue for m in metrics)
        
        if stripe_total > paypal_total:
            top_method = "Stripe"
        elif paypal_total > stripe_total:
            top_method = "PayPal"
        else:
            top_method = "Equal"
        
        return {
            "total_revenue": total_revenue,
            "total_payments": total_payments,
            "average_daily_revenue": average_daily_revenue,
            "growth_rate": growth_rate,
            "top_payment_method": top_method,
            "stripe_revenue": stripe_total,
            "paypal_revenue": paypal_total
        }
    
    # ==================== USER ANALYTICS ====================
    
    def update_user_analytics(self, user_id: int) -> UserAnalytics:
        """Aktualisiere User Analytics"""
        try:
            # Hole bestehende Analytics oder erstelle neue
            analytics = self.session.exec(
                select(UserAnalytics).where(UserAnalytics.user_id == user_id)
            ).first()
            
            if not analytics:
                analytics = UserAnalytics(user_id=user_id)
                self.session.add(analytics)
            
            # Hole User-Events
            events = self.get_events(user_id=user_id)
            
            # Berechne Metriken
            analytics.total_page_views = len([e for e in events if e.event_type == EventType.PAGE_VIEW])
            analytics.total_searches = len([e for e in events if e.event_type == EventType.SEARCH_PERFORMED])
            analytics.total_listings_created = len([e for e in events if e.event_type == EventType.LISTING_CREATED])
            analytics.total_listings_viewed = len([e for e in events if e.event_type == EventType.LISTING_VIEWED])
            analytics.total_chat_messages = len([e for e in events if e.event_type == EventType.CHAT_MESSAGE])
            analytics.total_favorites = len([e for e in events if e.event_type == EventType.FAVORITE_ADDED])
            
            # Zeit-Metriken
            if events:
                analytics.first_activity = min(e.created_at for e in events)
                analytics.last_activity = max(e.created_at for e in events)
                
                # Session-Zeit berechnen (vereinfacht)
                analytics.total_session_time = len(events) * 300  # 5 Min pro Event
            
            # Conversion Rate
            payments = self.session.exec(
                select(Payment).where(
                    and_(
                        Payment.user_id == user_id,
                        Payment.status == PaymentStatus.COMPLETED
                    )
                )
            ).all()
            
            if analytics.total_page_views > 0:
                analytics.conversion_rate = (len(payments) / analytics.total_page_views) * 100
            
            # Lifetime Value
            analytics.lifetime_value = sum(p.amount for p in payments)
            
            # Device/Browser (vereinfacht)
            if events and events[0].user_agent:
                user_agent = events[0].user_agent.lower()
                if 'mobile' in user_agent:
                    analytics.primary_device = 'mobile'
                elif 'tablet' in user_agent:
                    analytics.primary_device = 'tablet'
                else:
                    analytics.primary_device = 'desktop'
                
                if 'chrome' in user_agent:
                    analytics.primary_browser = 'chrome'
                elif 'firefox' in user_agent:
                    analytics.primary_browser = 'firefox'
                elif 'safari' in user_agent:
                    analytics.primary_browser = 'safari'
                else:
                    analytics.primary_browser = 'other'
            
            analytics.updated_at = datetime.utcnow()
            
            self.session.add(analytics)
            self.session.commit()
            self.session.refresh(analytics)
            
            return analytics
            
        except Exception as e:
            logger.error(f"Failed to update user analytics for user {user_id}: {e}")
            self.session.rollback()
            raise
    
    def get_user_analytics(self, user_id: int) -> Optional[UserAnalytics]:
        """Hole User Analytics"""
        return self.session.exec(
            select(UserAnalytics).where(UserAnalytics.user_id == user_id)
        ).first()
    
    def get_top_users(self, limit: int = 10) -> List[UserAnalytics]:
        """Hole Top User nach Engagement"""
        return self.session.exec(
            select(UserAnalytics)
            .order_by(UserAnalytics.total_page_views.desc())
            .limit(limit)
        ).all()
    
    # ==================== A/B TESTING ====================
    
    def create_ab_test(
        self,
        name: str,
        description: str,
        variants: Dict[str, Any],
        traffic_allocation: Dict[str, float],
        primary_metric: str,
        secondary_metrics: Optional[List[str]] = None
    ) -> ABTest:
        """Erstelle A/B Test"""
        try:
            ab_test = ABTest(
                name=name,
                description=description,
                variants=variants,
                traffic_allocation=traffic_allocation,
                primary_metric=primary_metric,
                secondary_metrics=secondary_metrics or []
            )
            
            self.session.add(ab_test)
            self.session.commit()
            self.session.refresh(ab_test)
            
            logger.info(f"A/B Test created: {name}")
            return ab_test
            
        except Exception as e:
            logger.error(f"Failed to create A/B test {name}: {e}")
            self.session.rollback()
            raise
    
    def assign_user_to_test(self, user_id: int, test_id: str) -> str:
        """Weise User einem Test-Variant zu"""
        try:
            # Prüfe ob User bereits zugewiesen
            existing = self.session.exec(
                select(ABTestAssignment).where(
                    and_(
                        ABTestAssignment.user_id == user_id,
                        ABTestAssignment.test_id == test_id
                    )
                )
            ).first()
            
            if existing:
                return existing.variant
            
            # Hole Test
            test = self.session.exec(
                select(ABTest).where(ABTest.test_id == test_id)
            ).first()
            
            if not test or test.status != "running":
                raise ValueError("Test not found or not running")
            
            # Bestimme Variant basierend auf Traffic Allocation
            import random
            random.seed(user_id)  # Konsistente Zuweisung
            rand = random.random()
            
            cumulative = 0
            for variant, allocation in test.traffic_allocation.items():
                cumulative += allocation
                if rand <= cumulative:
                    assignment = ABTestAssignment(
                        user_id=user_id,
                        test_id=test_id,
                        variant=variant
                    )
                    
                    self.session.add(assignment)
                    self.session.commit()
                    
                    return variant
            
            # Fallback
            variant = list(test.traffic_allocation.keys())[0]
            assignment = ABTestAssignment(
                user_id=user_id,
                test_id=test_id,
                variant=variant
            )
            
            self.session.add(assignment)
            self.session.commit()
            
            return variant
            
        except Exception as e:
            logger.error(f"Failed to assign user {user_id} to test {test_id}: {e}")
            self.session.rollback()
            raise
    
    def get_test_results(self, test_id: str) -> Dict[str, Any]:
        """Hole A/B Test Ergebnisse"""
        try:
            test = self.session.exec(
                select(ABTest).where(ABTest.test_id == test_id)
            ).first()
            
            if not test:
                raise ValueError("Test not found")
            
            # Hole Assignments
            assignments = self.session.exec(
                select(ABTestAssignment).where(ABTestAssignment.test_id == test_id)
            ).all()
            
            # Gruppiere nach Variants
            variant_stats = {}
            for variant in test.variants.keys():
                variant_assignments = [a for a in assignments if a.variant == variant]
                
                variant_stats[variant] = {
                    "total_users": len(variant_assignments),
                    "converted_users": len([a for a in variant_assignments if a.converted]),
                    "conversion_rate": 0,
                    "total_value": sum(a.conversion_value or 0 for a in variant_assignments)
                }
                
                if variant_stats[variant]["total_users"] > 0:
                    variant_stats[variant]["conversion_rate"] = (
                        variant_stats[variant]["converted_users"] / 
                        variant_stats[variant]["total_users"]
                    ) * 100
            
            # Bestimme Winner
            winner = max(
                variant_stats.keys(),
                key=lambda v: variant_stats[v]["conversion_rate"]
            )
            
            # Speichere Ergebnisse
            test.results = variant_stats
            test.winner = winner
            test.updated_at = datetime.utcnow()
            
            self.session.add(test)
            self.session.commit()
            
            return {
                "test_id": test_id,
                "test_name": test.name,
                "status": test.status,
                "variants": variant_stats,
                "winner": winner,
                "confidence_level": 95.0  # Vereinfacht
            }
            
        except Exception as e:
            logger.error(f"Failed to get test results for {test_id}: {e}")
            self.session.rollback()
            raise

File: frontend/src/services/analyticsService.ts
/**
 * Analytics Service für Business Intelligence
 */

// API Base URL
const API_BASE_URL = 'http://localhost:8000';

// Helper function für API calls
async function apiCall(endpoint: string, options: RequestInit = {}) {
  const token = localStorage.getItem('token');
  
  const response = await fetch(`${API_BASE_URL}${endpoint}`, {
    headers: {
      'Content-Type': 'application/json',
      ...(token && { 'Authorization': `Bearer ${token}` }),
      ...options.headers,
    },
    ...options,
  });
  
  if (!response.ok) {
    if (response.status === 401) {
      // Token ungültig - User zur Login-Seite weiterleiten
      localStorage.removeItem('token');
      window.location.href = '/login';
      throw new Error('401 Unauthorized');
    }
    throw new Error(`API call failed: ${response.status}`);
  }
  
  return response.json();
}

// ==================== TYPES ====================

export interface AnalyticsEvent {
  id: number;
  event_id: string;
  event_type: string;
  user_id?: number;
  session_id?: string;
  properties?: Record<string, any>;
  value?: number;
  ip_address?: string;
  user_agent?: string;
  referrer?: string;
  created_at: string;
  event_date: string;
}

export interface RevenueMetric {
  id: number;
  metric_id: string;
  date: string;
  hour?: number;
  total_revenue: number;
  payment_count: number;
  subscription_revenue: number;
  one_time_revenue: number;
  stripe_revenue: number;
  paypal_revenue: number;
  new_users: number;
  active_users: number;
  paying_users: number;
  created_at: string;
  updated_at: string;
}

export interface UserAnalytics {
  id: number;
  user_id: number;
  total_page_views: number;
  total_searches: number;
  total_listings_created: number;
  total_listings_viewed: number;
  total_chat_messages: number;
  total_favorites: number;
  first_activity?: string;
  last_activity?: string;
  total_session_time: number;
  conversion_rate: number;
  lifetime_value: number;
  primary_device?: string;
  primary_browser?: string;
  created_at: string;
  updated_at: string;
}

export interface ABTest {
  id: number;
  test_id: string;
  name: string;
  description?: string;
  status: string;
  variants: Record<string, any>;
  traffic_allocation: Record<string, number>;
  primary_metric: string;
  secondary_metrics?: string[];
  start_date?: string;
  end_date?: string;
  results?: Record<string, any>;
  winner?: string;
  confidence_level?: number;
  created_at: string;
  updated_at: string;
}

export interface DashboardOverview {
  revenue: {
    total_revenue: number;
    total_payments: number;
    average_daily_revenue: number;
    growth_rate: number;
    top_payment_method: string;
    stripe_revenue: number;
    paypal_revenue: number;
  };
  top_users: UserAnalytics[];
  recent_activity: {
    total_events: number;
    event_types: Record<string, number>;
  };
  period_days: number;
}

export interface RealTimeMetrics {
  today_revenue: number;
  yesterday_revenue: number;
  revenue_growth: number;
  recent_events: number;
  active_users: number;
  timestamp: string;
}

// ==================== ANALYTICS SERVICE ====================

class AnalyticsService {
  /**
   * Tracke ein Event
   */
  async trackEvent(
    eventType: string,
    userId?: number,
    sessionId?: string,
    properties?: Record<string, any>,
    value?: number
  ): Promise<{ success: boolean; event_id: string; message: string }> {
    try {
      const response = await apiCall('/api/analytics/events/track', {
        method: 'POST',
        body: JSON.stringify({
          event_type: eventType,
          user_id: userId,
          session_id: sessionId,
          properties: properties,
          value: value
        }),
      });
      return response;
    } catch (error) {
      console.error('Event tracking failed:', error);
      throw new Error('Event konnte nicht getrackt werden');
    }
  }

  /**
   * Hole Events
   */
  async getEvents(params?: {
    eventTypes?: string[];
    userId?: number;
    startDate?: string;
    endDate?: string;
    limit?: number;
  }): Promise<{ success: boolean; events: AnalyticsEvent[]; count: number }> {
    try {
      const queryParams = new URLSearchParams();
      if (params?.eventTypes) {
        params.eventTypes.forEach(type => queryParams.append('event_types', type));
      }
      if (params?.userId) queryParams.append('user_id', params.userId.toString());
      if (params?.startDate) queryParams.append('start_date', params.startDate);
      if (params?.endDate) queryParams.append('end_date', params.endDate);
      if (params?.limit) queryParams.append('limit', params.limit.toString());

      const response = await apiCall(`/api/analytics/events?${queryParams.toString()}`);
      return response;
    } catch (error) {
      console.error('Events retrieval failed:', error);
      throw new Error('Events konnten nicht abgerufen werden');
    }
  }

  /**
   * Berechne Tagesumsatz
   */
  async calculateDailyRevenue(targetDate: string): Promise<{ success: boolean; metric: RevenueMetric; message: string }> {
    try {
      const response = await apiCall(`/api/analytics/revenue/calculate/${targetDate}`, {
        method: 'POST',
      });
      return response;
    } catch (error) {
      console.error('Revenue calculation failed:', error);
      throw new Error('Umsatz konnte nicht berechnet werden');
    }
  }

  /**
   * Hole Revenue-Metriken
   */
  async getRevenueMetrics(params?: {
    startDate?: string;
    endDate?: string;
    days?: number;
  }): Promise<{ success: boolean; metrics: RevenueMetric[]; period: any }> {
    try {
      const queryParams = new URLSearchParams();
      if (params?.startDate) queryParams.append('start_date', params.startDate);
      if (params?.endDate) queryParams.append('end_date', params.endDate);
      if (params?.days) queryParams.append('days', params.days.toString());

      const response = await apiCall(`/api/analytics/revenue/metrics?${queryParams.toString()}`);
      return response;
    } catch (error) {
      console.error('Revenue metrics retrieval failed:', error);
      throw new Error('Revenue-Metriken konnten nicht abgerufen werden');
    }
  }

  /**
   * Hole Revenue-Zusammenfassung
   */
  async getRevenueSummary(days: number = 30): Promise<{ success: boolean; summary: any; period_days: number }> {
    try {
      const response = await apiCall(`/api/analytics/revenue/summary?days=${days}`);
      return response;
    } catch (error) {
      console.error('Revenue summary retrieval failed:', error);
      throw new Error('Revenue-Zusammenfassung konnte nicht abgerufen werden');
    }
  }

  /**
   * Aktualisiere User Analytics
   */
  async updateUserAnalytics(userId: number): Promise<{ success: boolean; analytics: UserAnalytics; message: string }> {
    try {
      const response = await apiCall(`/api/analytics/users/${userId}/update`, {
        method: 'POST',
      });
      return response;
    } catch (error) {
      console.error('User analytics update failed:', error);
      throw new Error('User Analytics konnten nicht aktualisiert werden');
    }
  }

  /**
   * Hole User Analytics
   */
  async getUserAnalytics(userId: number): Promise<{ success: boolean; analytics: UserAnalytics }> {
    try {
      const response = await apiCall(`/api/analytics/users/${userId}`);
      return response;
    } catch (error) {
      console.error('User analytics retrieval failed:', error);
      throw new Error('User Analytics konnten nicht abgerufen werden');
    }
  }

  /**
   * Hole Top User
   */
  async getTopUsers(limit: number = 10): Promise<{ success: boolean; top_users: UserAnalytics[]; count: number }> {
    try {
      const response = await apiCall(`/api/analytics/users/top?limit=${limit}`);
      return response;
    } catch (error) {
      console.error('Top users retrieval failed:', error);
      throw new Error('Top User konnten nicht abgerufen werden');
    }
  }

  /**
   * Erstelle A/B Test
   */
  async createABTest(
    name: string,
    description: string,
    variants: Record<string, any>,
    trafficAllocation: Record<string, number>,
    primaryMetric: string,
    secondaryMetrics?: string[]
  ): Promise<{ success: boolean; ab_test: ABTest; message: string }> {
    try {
      const response = await apiCall('/api/analytics/ab-tests', {
        method: 'POST',
        body: JSON.stringify({
          name,
          description,
          variants,
          traffic_allocation: trafficAllocation,
          primary_metric: primaryMetric,
          secondary_metrics: secondaryMetrics
        }),
      });
      return response;
    } catch (error) {
      console.error('A/B test creation failed:', error);
      throw new Error('A/B Test konnte nicht erstellt werden');
    }
  }

  /**
   * Weise User einem Test zu
   */
  async assignUserToTest(testId: string, userId: number): Promise<{ success: boolean; variant: string; message: string }> {
    try {
      const response = await apiCall(`/api/analytics/ab-tests/${testId}/assign/${userId}`, {
        method: 'POST',
      });
      return response;
    } catch (error) {
      console.error('User assignment failed:', error);
      throw new Error('User konnte nicht zugewiesen werden');
    }
  }

  /**
   * Hole Test Ergebnisse
   */
  async getTestResults(testId: string): Promise<{ success: boolean; results: any }> {
    try {
      const response = await apiCall(`/api/analytics/ab-tests/${testId}/results`);
      return response;
    } catch (error) {
      console.error('Test results retrieval failed:', error);
      throw new Error('Test-Ergebnisse konnten nicht abgerufen werden');
    }
  }

  /**
   * Hole alle A/B Tests
   */
  async getABTests(status?: string): Promise<{ success: boolean; ab_tests: ABTest[]; count: number }> {
    try {
      const queryParams = status ? `?status=${status}` : '';
      const response = await apiCall(`/api/analytics/ab-tests${queryParams}`);
      return response;
    } catch (error) {
      console.error('A/B tests retrieval failed:', error);
      throw new Error('A/B Tests konnten nicht abgerufen werden');
    }
  }

  /**
   * Hole Dashboard-Übersicht
   */
  async getDashboardOverview(days: number = 30): Promise<{ success: boolean; overview: DashboardOverview }> {
    try {
      const response = await apiCall(`/api/analytics/dashboard/overview?days=${days}`);
      return response;
    } catch (error) {
      console.error('Dashboard overview retrieval failed:', error);
      throw new Error('Dashboard-Übersicht konnte nicht abgerufen werden');
    }
  }

  /**
   * Hole Echtzeit-Metriken
   */
  async getRealTimeMetrics(): Promise<{ success: boolean; real_time: RealTimeMetrics }> {
    try {
      const response = await apiCall('/api/analytics/dashboard/real-time');
      return response;
    } catch (error) {
      console.error('Real-time metrics retrieval failed:', error);
      throw new Error('Echtzeit-Metriken konnten nicht abgerufen werden');
    }
  }
}

// Export singleton instance
export const analyticsService = new AnalyticsService();
export default analyticsService;
